packages/navigator/src/browser/navigator-contribution.ts

```diff
@@ -161,6 +192,24 @@ export class FileNavigatorContribution extends AbstractViewContribution<FileNavi
         };
         updateFocusContextKeys();
         this.shell.activeChanged.connect(updateFocusContextKeys);
+        this.workspaceCommandContribution.onDidCreateNewFile(async uri => this.onDidCreateNewResource(uri));
+        this.workspaceCommandContribution.onDidCreateNewFolder(async uri => this.onDidCreateNewResource(uri));
+    }
+
+    private async onDidCreateNewResource(uri: URI): Promise<void> {
+        const navigator = await this.tryGetWidget();
+        if (!navigator || !navigator.isVisible) {
+            return;
+        }
+        const model: FileNavigatorModel = navigator.model;
+        const parent = await model.revealFile(uri.parent);
```

Comment:
it does not work for composite path, try to enter following folder: `foo/bar/baz`.

I think we should pass original parent as the part of `DidCreateNew[File|Folder]` events.
```ts
export interface DidCreateNewFileEvent {
    uri: URI
    parent: URI
}
```

Reply:
done